pipeline/trusted-resources/cmd/sign/main.go

```diff
@@ -32,18 +32,22 @@ import (
 	"github.com/sigstore/sigstore/pkg/signature/kms"
 	"github.com/tektoncd/experimental/pipelines/trusted-resources/pkg/trustedtask"
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/webhook/resourcesemantics"
 	"sigs.k8s.io/yaml"
 )
 
 var (
-	cosignKey  = flag.String("ck", "", "cosign private key path")
-	kmsKey     = flag.String("kms", "", "kms key path")
-	taskFile   = flag.String("ts", "", "YAML file path for tekton task")
-	targetDir  = flag.String("td", "", "Dir to save the signed files")
-	targetFile = flag.String("tf", "signed.yaml", "Filename of the signed file")
+	cosignKey    = flag.String("ck", "", "cosign private key path")
+	kmsKey       = flag.String("kms", "", "kms key path")
+	resourceFile = flag.String("rf", "", "YAML file path for tekton resources")
+	kind         = flag.String("kd", "Task", "The kind of the signed object")
+	format       = flag.Bool("fmt", true, "Whether we add missing values for the CRD")
+	targetDir    = flag.String("td", "", "Dir to save the signed files")
+	targetFile   = flag.String("tf", "signed.yaml", "Filename of the signed file")
 )
 
-// This is a demo of how to generate signed task files
+// This is a demo of how to generate signed task or pipeline files
 func main() {
```

Comment:
Super nit: the current code structure is okay for now, but if we want to move it out from the experimental repo in the future, we'd better to keep the implementation details outside of the main function, for example:

```
signer, err := getSigner(...)
crd, err := getCRD(resourceFile)
formatCRD(&crd)
signCRD(...)
```

Reply:
yes! (I didn't do this because I want to avoid more unit tests. No I'm just kidding. 😄 )